### PR TITLE
Linq queries by Foxtra

### PR DESCRIPTION
**Pull request by Foxtra**
Hello,

please check my pull request with homework 5, when you have time.
Sorry for delay.

Regards,
Sergei

### CODE REVIEW

**Comment by Foxtra** [comment]
Task link: #171

File: HW5/Sergey_Tkachev/LinqQueries/Task/Task/LinqSamples.cs [hunk 1]
```diff
@@ -0,0 +1,476 @@
...
+            foreach (var p in products)
+            {
+                ObjectDumper.Write(p.Category);
+                foreach (var s in p.inStockGroup)
+                {
+                    ObjectDumper.Write(s.inStock ? "In Stock" : "Out of Stock");
+                    foreach (var r in s.Price)
+                        ObjectDumper.Write("Price =  " + r.UnitPrice + " Product: " + r.ProductName);
+                }
+                ObjectDumper.Write(String.Empty);
```
Comment (vladimirchr) [inline HW5/Sergey_Tkachev/LinqQueries/Task/Task/LinqSamples.cs]: Despite the fact that this is not explicitly indicated in issuee, it is better to use `ObjectDumper()` like this
`foreach (var p in products)`
`{`
`ObjectDumper.Write(p, 2);`
`}`

File: HW5/Sergey_Tkachev/LinqQueries/Task/Task/LinqSamples.cs [hunk 2]
```diff
@@ -0,0 +1,476 @@
...
+                    "inside by avalability in stock, inside by price.")]
+
+        public void LinqA8()
+        {
+            int Cheap = 20;
+            int Expensive = 40;
+
+            var products = dataSource.Products
+                .OrderBy(p => p.UnitPrice)
+                .GroupBy(p => p.UnitPrice < Cheap ? "Cheap" : p.UnitPrice < Expensive ? "Average" : "Expensive");
```
Comment (vladimirchr) [inline HW5/Sergey_Tkachev/LinqQueries/Task/Task/LinqSamples.cs]: We need to add Select after `GroupBy()`:
`var products2 = dataSource.Products`
`.GroupBy(p => p.UnitPrice < Cheap ? "Cheap" : p.UnitPrice < Expensive ? "Average" : "Expensive")`
`.Select(x => new { Price_Category = x.Key, Product = x`
`.Select(y => new { y.ProductName, y.UnitPrice }) });`
ObjectDump

File: HW5/Sergey_Tkachev/LinqQueries/Task/Task/LinqSamples.cs [hunk 3]
```diff
@@ -0,0 +1,476 @@
...
+        [Category("LinqToSQL")]
+        [Title("Task B.1.1")]
+        [Description("This query returns all clients, who have total turnover over 9")]
+        public void LinqB11()
+        {
+            int threshold = 9;
+
+            var clients =
+                db.Customers
+                    .Where(o => o.Orders.Count() > threshold)
```
Comment (vladimirchr) [inline HW5/Sergey_Tkachev/LinqQueries/Task/Task/LinqSamples.cs]: We should implement the task in the same way for both: the first case and the database case
Can we do this as in the first case?
`var clients = dataSource.Customers.Where(c => (c.Orders.Sum(o => o.Total) > threshold));`

File: HW5/Sergey_Tkachev/LinqQueries/Task/Task/LinqSamples.cs [hunk 4]
```diff
@@ -0,0 +1,476 @@
...
+            }
+        }
+
+        [Category("LinqToSQL")]
+        [Title("Task B.3")]
+        [Description("This query returns all clients, who have orders worth > 5000")]
+
+        public void LinqB3()
+        {
+            decimal someValueX = 5000;
```
Comment (vladimirchr) [inline HW5/Sergey_Tkachev/LinqQueries/Task/Task/LinqSamples.cs]: We could use a more meaningful name for better redability like `minOrderValue,` `singleOrderThreshold`

**Pull request merged**